definitions/reports/inventory.rb

```diff
@@ -0,0 +1,92 @@
+# copy the file and add the .rb suffix
+module Checks
+  module Report
+    class Inventory < ForemanMaintain::Report
+      metadata do
+        description 'Facts about hosts and the rest of the inventory'
+      end
+
+      # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+      def run
+        # Hosts
+        hosts_by_type_count =
+          feature(:foreman_database).
+          query("select type, count(*) from hosts group by type").
+          to_h { |row| [(row['type'] || '').sub('Host::', ''), row['count'].to_i] }
+
+        # OS usage
+        hosts_by_os_count =
+          feature(:foreman_database).
+          query(
+            <<-SQL
+              select max(operatingsystems.name) as os_name, count(*) as hosts_count
+              from hosts inner join operatingsystems on operatingsystem_id = operatingsystems.id
+              group by operatingsystem_id
+            SQL
+          ).
+          to_h { |row| [row['os_name'], row['hosts_count'].to_i] }
+
+        # Facts usage
+        facts_by_type =
+          feature(:foreman_database).
+          query(
+            <<-SQL
+              select fact_names.type,
+                     min(fact_values.updated_at) as min_update_time,
+                     max(fact_values.updated_at) as max_update_time,
+                     count(fact_values.id) as values_count
+              from fact_values inner join fact_names on fact_name_id = fact_names.id
+              group by fact_names.type
+            SQL
+          ).
+          to_h { |row| [row['type'].sub('FactName', ''), to_fact_hash(row)] }
```

Comment:
Be nice if each of these chunks of data were their own method instead of one giant method.